File: December 2024 Leetcode Solution/December-04.py
class Solution:
    def canMakeSubsequence(self, str1: str, str2: str) -> bool:
        j, m = 0, len(str2)
        for c in str1:
            if j < m and (c == str2[j] or chr((ord(c) - ord('a') + 1) % 26 + ord('a')) == str2[j]):
                j += 1
                if j == m:
                    return True
        return j == m

File: December 2024 Leetcode Solution/December-04.rs
impl Solution {
    pub fn can_make_subsequence(str1: String, str2: String) -> bool {
        let (mut j, m) = (0, str2.len());
        let str2_bytes = str2.as_bytes();
        for c in str1.bytes() {
            if j < m && (c == str2_bytes[j] || (c - b'a' + 1) % 26 + b'a' == str2_bytes[j]) {
                j += 1;
                if j == m {
                    return true;
                }
            }
        }
        j == m
    }
}
